File: student/src/main/java/com/studentdashboard/service/AcademicService.java
package com.studentdashboard.service;

import com.studentdashboard.dto.request.SubjectRequest;
import com.studentdashboard.dto.request.MarksRequest;
import com.studentdashboard.dto.request.AttendanceRequest;
import com.studentdashboard.entity.Subject;
import com.studentdashboard.entity.StudentMarks;
import com.studentdashboard.entity.Attendance;

import java.util.List;

public interface AcademicService {
    Subject addSubject(String username, SubjectRequest request);
    void deleteSubject(String username, Long subjectId);
    List<Subject> getSubjects(String username);
    List<Subject> getSubjectsBySemester(String username, String semester);
    
    StudentMarks addOrUpdateMarks(String username, MarksRequest request);
    List<StudentMarks> getMarks(String username);
    
    Attendance addOrUpdateAttendance(String username, AttendanceRequest request);
    List<Attendance> getAttendance(String username);
    
    void recalculateAnalytics(String username);
}

File: src/services/academicService.js
import api from './api';

export const academicService = {
  getSubjects: async () => {
    const res = await api.get('/academic/subjects');
    return res.data;
  },
  
  getSubjectsBySemester: async (semester) => {
    const res = await api.get(`/academic/subjects/${semester}`);
    return res.data;
  },
  
  addSubject: async (subjectData) => {
    const res = await api.post('/academic/subject', subjectData);
    return res.data;
  },
  
  deleteSubject: async (id) => {
    const res = await api.delete(`/academic/subject/${id}`);
    return res.data;
  },
  
  getMarks: async () => {
    const res = await api.get('/academic/marks');
    return res.data;
  },
  
  addOrUpdateMarks: async (marksData) => {
    const res = await api.post('/academic/marks', marksData);
    return res.data;
  },
  
  getAttendance: async () => {
    const res = await api.get('/academic/attendance');
    return res.data;
  },
  
  addOrUpdateAttendance: async (attendanceData) => {
    const res = await api.post('/academic/attendance', attendanceData);
    return res.data;
  }
};

export default academicService;
